projects/charger/src/charger_pin.c

```diff
@@ -0,0 +1,49 @@
+#include "charger_pin.h"
+
+#include <stddef.h>
+
+#include "charger_events.h"
+#include "event_queue.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt_def.h"
+#include "status.h"
+
+static void prv_charger_pin_it(const GPIOAddress *address, void *context) {
+  (void)context;
+  GPIOState state = NUM_GPIO_STATES;
+  gpio_get_state(address, &state);
+  switch (state) {
```

Comment:
Seems a little overkill for a pretty simple function. :P

Reply:
Haha, yeah I suppose so...